types/test.d.ts

```diff
@@ -0,0 +1,203 @@
+  looseIs: Assertions.Equal
+
+  notSame: Assertions.NotEqual
+  inequivalent: Assertions.NotEqual
+  looseInequal: Assertions.NotEqual
+  notDeep: Assertions.NotEqual
+  deepInequal: Assertions.NotEqual
+  notLoose: Assertions.NotEqual
+  looseNot: Assertions.NotEqual
+
+  strictSame: Assertions.Equal
+  strictEquivalent: Assertions.Equal
+  strictDeepEqual: Assertions.Equal
+  sameStrict: Assertions.Equal
+  deepIs: Assertions.Equal
+  isDeeply: Assertions.Equal
+  isDeep: Assertions.Equal
+  strictDeepEquals: Assertions.Equal
+
+  strictNotSame: Assertions.NotEqual
+  strictInequivalent: Assertions.NotEqual
+  strictDeepInequal: Assertions.NotEqual
+  notSameStrict: Assertions.NotEqual
+  deepNot: Assertions.NotEqual
+  notDeeply: Assertions.NotEqual
+  strictDeepInequals: Assertions.NotEqual
+  notStrictSame: Assertions.NotEqual
+
+  match: Assertions.Match
+  has: Assertions.Match
+  hasFields: Assertions.Match
+  matches: Assertions.Match
+  similar: Assertions.Match
+  like: Assertions.Match
+  isLike: Assertions.Match
+  includes: Assertions.Match
+  include: Assertions.Match
+  contains: Assertions.Match
+
+  notMatch: Assertions.Match
+  dissimilar: Assertions.Match
+  unsimilar: Assertions.Match
+  notSimilar: Assertions.Match
+  unlike: Assertions.Match
+  isUnlike: Assertions.Match
+  notLike: Assertions.Match
+  isNotLike: Assertions.Match
+  doesNotHave: Assertions.Match
+  isNotSimilar: Assertions.Match
+  isDissimilar: Assertions.Match
+
+  type: Assertions.Type
+  isa: Assertions.Type
+  isA: Assertions.Type
+}
+
+export declare namespace Options {
+  export interface Bag {
+    [key: string]: any
+  }
+
+  export interface Pragma {
+    [key: string]: boolean
+  }
+
+  export interface Assert extends Bag {
+    todo?: boolean | string,
+    skip?: boolean | string,
+    diagnostic?: boolean
+  }
+
+  export interface Spawn extends Assert {
+    bail?: boolean,
+    timeout?: number
+  }
+
+  export interface Test extends Assert {
+    timeout?: number,
+    bail?: boolean,
+    autoend?: boolean,
+    buffered?: boolean,
+    jobs?: number,
+    grep?: RegExp[],
+    only?: boolean,
+    runOnly?: boolean
+  }
+}
+
+export declare namespace Assertions {
+  export type Basic = (obj: any, message?: string, extra?: Options.Assert) => boolean
+
+  export type Throws = (fn?: (a: any) => any, expectedError?: Error, message?: string, extra?: Options.Assert) => boolean
+
+  export type DoesNotThrow = (fn?: (a: any) => any, message?: string, extra?: Options.Assert) => boolean
+
+  export type Equal = (found: any, wanted: any, message?: string, extra?: Options.Assert) => boolean
+
+  export type NotEqual = (found: any, notWanted: any, message?: string, extra?: Options.Assert) => boolean
+
+  export type Match = (found: any, pattern: any, message?: string, extra?: Options.Assert) => boolean
+
+  export type Type = (found: any, type: string | ((a: any) => any), message?: string, extra?: Options.Assert) => boolean
```

Comment:
The type should be `string` or a [constructor function](https://www.simonholywell.com/post/typescript-constructor-type.html):

```suggestion
  type Constructor<T extends {} = {}> = new (...args: any[]) => T
  export type Type = (found: any, type: string | Constructor, message?: string, extra?: Options.Assert) => boolean
```